Authenticators/Elkarte/elkarteauth.py

```diff
@@ -0,0 +1,887 @@
+            and makes sure an authenticator gets attached if needed.
+            """
+            if not cfg.murmur.servers or server.id() in cfg.murmur.servers:
+                info('Setting authenticator for virtual server %d', server.id())
+                try:
+                    server.setAuthenticator(app.auth)
+                # Apparently this server was restarted without us noticing
+                except (Murmur.InvalidSecretException, Ice.UnknownUserException), e:
+                    if hasattr(e, "unknown") and e.unknown != "Murmur::InvalidSecretException":
+                        # Special handling for Murmur 1.2.2 servers with invalid slice files
+                        raise e
+                    
+                    error('Invalid ice secret')
+                    return
+            else:
+                debug('Virtual server %d got started', server.id())
+
+        @fortifyIceFu()
+        @checkSecret
+        def stopped(self, server, current = None):
+            """
+            This function is called when a virtual server is stopped
+            """
+            if self.app.connected:
+                # Only try to output the server id if we think we are still connected to prevent
+                # flooding of our thread pool
+                try:
+                    if not cfg.murmur.servers or server.id() in cfg.murmur.servers:
+                        info('Authenticated virtual server %d got stopped', server.id())
+                    else:
+                        debug('Virtual server %d got stopped', server.id())
+                    return
+                except Ice.ConnectionRefusedException:
+                    self.app.connected = False
+            
+            debug('Server shutdown stopped a virtual server')
+    
+    if cfg.user.reject_on_error: # Python 2.4 compat
+        authenticateFortifyResult = (-1, None, None)
+    else:
+        authenticateFortifyResult = (-2, None, None)
+        
+    class elkarteauthenticator(Murmur.ServerUpdatingAuthenticator):
+        texture_cache = {}
+        def __init__(self):
+            Murmur.ServerUpdatingAuthenticator.__init__(self)
+
+        @fortifyIceFu(authenticateFortifyResult)
+        @checkSecret
+        def authenticate(self, name, pw, certlist, certhash, strong, current = None):
+            """
+            This function is called to authenticate a user
+            """
+            
+            # Search for the user in the database
+            FALL_THROUGH = -2
+            AUTH_REFUSED = -1
+            
+            if name == 'SuperUser':
+                debug('Forced fall through for SuperUser')
+                return (FALL_THROUGH, None, None)
+            
+            try:
+                sql = 'SELECT id_member, passwd, id_group, member_name, real_name, additional_groups, is_activated FROM %smembers WHERE LOWER(member_name) = LOWER(%%s)' % cfg.database.prefix
+                cur = threadDB.execute(sql, [name])
```

Comment:
Good to know. This might very well what is causing the error in #5 . It worked when I wrote it so I might have been unintentionally relying on undocumented behaviors or it was deprecated in the meantime.